File: contest/joined/TPR-21/B.py
# Author : ysh
# 05/29/2022 Sun 15:05:47.33
from tkinter import E
from numpy import gcd


a = [int(i) for i in input().split()]
c = a[0]
d = a[1]
if c > d:
    e = c
    c = d
    d = e
if d - c == 1:
    print(-1)
    quit()
print(d - c)
quit()

File: contest/joined/TPR-21/B.cpp
// Author : ysh
// 05/29/2022 Sun 15:13:27.30
#include<bits/stdc++.h>
using namespace std;
int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);

    int a,b;cin>>a>>b;
    if(a > b) {
        swap(a,b);
    }
    // cout<<(b - a == 1 ? -1 : b - a);
    // return 0;
    if(a == 0 && b == 0) {
        cout<<-1;
        return 0;
    }
    if(a == 0) {
        cout<<b;
        return 0;
    }
    for(int i = 2;i<b;i++) {
        if(a % i == b % i) {
            cout<<i;
            return 0;
        }
    }
    cout<<-1;
    return 0;
}